src/core/partition.h

```diff
@@ -0,0 +1,173 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    The partitioned storage for global library state. The partitioning allows
+    multiple threads to operate on the library simultaneously with minimal
+    contention.
+
+    The primary goal of partitioning is to allow multiple threads to allocate
+    and free pool memory simultaneously without contention. It also maintains
+    isolation for other state that may be commonly accessed by multiple threads,
+    such as performance counters and stateless resets and retries.ABC
+
+    A partition is always (soft) affinitized to a specific processor. By default,
+    partitions are one to one with processors. Though, an application may choose
+    to create partitions on a subset of processors. In this case, the partition
+    will be shared by all processors that are not explicitly affinitized to a
+    partition, though generally the library will try to only execute on those
+    specified processors.
```

Comment:
I am not sure I understand that sentence properly.
Does it mean that one partition can be associated to multiple processors if the application say so? Or that the application can create less partition than the number of processor (but each partition is still associated to one processor)?
If I have 4 CPU and create 2 partitions only as an application, what does it mean for partition sharing between CPU?

Reply:
So if an app creates two partitions targeting CPUs A and B, and then submits a request on CPU C, the QUIC library will pick one of the two previously created partitions (neither matching CPU C) and implicitly associate the request to that partition? Or is there a catch-all partition for all remaining processors?

Reply:
Yes, it will try to pick a close partition and assign the connection to that one. RSS may later move it again.